libvips/foreign/pngsave.c

```diff
@@ -133,13 +136,37 @@ vips_foreign_save_png_class_init( VipsForeignSavePngClass *class )
 		VIPS_TYPE_FOREIGN_PNG_FILTER,
 		VIPS_FOREIGN_PNG_FILTER_ALL );
 
+	VIPS_ARG_INT( class, "colors", 13,
+		_( "Colors" ),
+		_( "Number of palette entries" ),
+		VIPS_ARGUMENT_OPTIONAL_INPUT,
+		G_STRUCT_OFFSET( VipsForeignSavePng, colors ),
+		2, 256, 256 );
+
+	VIPS_ARG_INT( class, "Q", 14,
+		_( "Q" ),
+		_( "Q factor" ),
+		VIPS_ARGUMENT_OPTIONAL_INPUT,
+		G_STRUCT_OFFSET( VipsForeignSavePng, Q ),
+		0, 100, 75 );
```

Comment:
The libimagequant docs suggest it will fail if the desired "quality" can't be met. Its default for `Q` is `0`, which tells it to proceed regardless of quality. Should we match that default here to reduce the possibility of unexpected errors?

Reply:
In libimagequant there is a minimum and maximum quality. We always set the minimum quality to `0` so it never fails because quality can't be met. The `Q` parameter maps to maximum quality only.

It could be argued that the default quality should be `100` instead of `75`, but I chose that value to match other lossy compressors with the same option, like @jpegsave@.

However having a default of `75` also means that a call with @colors=256@ will likely produce less colors, unless `Q=100` is specified, so for predicting behavior `100` would be a better default.

Reply:
@lovell I'm happy to hear your thoughts on what#s the better default for `Q`here, I'm slightly leaning towards changing it to `100`, but I'm still undecided.

Reply:
Ah, I didn't spot that it was a range, in which case yes, I agree with a default of `100` here.

Reply:
I've changed the default for `Q` to `100` and also made the parameter description more useful.